### PR TITLE
T4193: initial implementation of the bridge firewall (conf mode)

### PR DESCRIPTION
**Pull request by dmbaturin**
## Change Summary

Initial implementation of the bridge firewal (conf mode)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4193

## Component(s) name

Firwewall.

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/conf_mode/firewall-bridge.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import vyos.util
+
+from vyos.config import Config
+from vyos.ifconfig import Section
+from vyos.util import call
+from vyos.util import cmd
+
+from vyos import ConfigError
+from vyos import airbag
+from pprint import pprint
```
Comment (sever-sever) [inline src/conf_mode/firewall-bridge.py]: It can be removed

File: src/conf_mode/firewall-bridge.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                 ['firewall'], key_mangling=('-', '_'),
+                 get_first_key=True,
+                 no_tag_node_value_mangle=True)
+
+    policy_dict = vyos.util.get_sub_dict(firewall, ['policy', 'bridge', 'forward'])
+    if not policy_dict:
+        return None
+
+
+    pprint(firewall)
```
Comment (sever-sever) [inline src/conf_mode/firewall-bridge.py]: Remove this `pprint`

**Review by sever-sever — CHANGES_REQUESTED** [review]

File: interface-definitions/firewall-bridge.xml.in [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+<interfaceDefinition>
+  <node name="firewall">
+    <children>
+      <node name="policy">
+        <properties>
+          <help>Firewall policy</help>
+        </properties>
+        <children>
+          <node name="bridge" owner="${vyos_conf_scripts_dir}/firewall-bridge.py"">
```
Comment (sever-sever) [inline interface-definitions/firewall-bridge.xml.in]: .py"">
Delete extra quote

**Review by sever-sever — CHANGES_REQUESTED** [review]

**Comment by sarthurdev** [comment]
firewall.py should be updated from current to include a few new bugfixes in `parse_rule()`

**Review by sever-sever — APPROVED** [review]

**Comment by andamasov** [comment]
@c-po @jestabro can you check this one, please

**Review by jestabro — APPROVED** [review]

**Comment by dmbaturin** [comment]
We shouldn't merge it until 1.3.2 is tagged off.

**Comment by sever-sever** [comment]
@dmbaturin Could you solve conflicts?
Will we merge it?

**Comment by dmbaturin** [comment]
The new firewall implementation is much more conductive to this change, and with 1.4 nearing the LTS phase, this is likely irrelevant.

**Pull request closed without merging**